.github/workflows/deploy-preview.yml

```diff
@@ -78,6 +81,16 @@ jobs:
           sparse-checkout: |
             deploy-preview
 
+      # sparse checkout with cone mode disabled includes root files, even when using exclusions
+      # see https://github.com/actions/checkout/issues/1430#issuecomment-1756326892
+      - name: Keep only necessary files
+        if: github.event.action == 'opened' || github.event.action == 'synchronize'
+        shell: bash
+        run: |
+          find . -mindepth 1 -not \( -path "./${{ inputs.source_directory }}" -o -path "./${{ inputs.source_directory }}/*" -o -path "./deploy-preview-files" -o -path "./deploy-preview-files/*" \) -exec rm -rf {} +
+          ls -al
+          ls -al $trimmed_value
+
```

Comment:
Can we also just consider trashing all files that aren't our docs directory? 

Or perhaps another way is to do all the checking out into the default workspace but then copying what we want into a new subdirectory.

The workspace tree may then look like:

```
.
├── build
│   ├── deploy-preview-files
│   ├── dist
│   └── docs
│       └── sources
├── deploy-preview-files
└── grafana
```

Reply:
In that previous comment, the `grafana` directory contains the grafana/grafana repo checkout (sparse or otherwise). The `build` directory is where we copy the `deploy-preview-files;` from the grafana/writers-toolkit checkout and `docs/sources/` from the grafana/grafana checkout.

We then switch to that `build` directory before running all the rest of the action

Reply:
Yep, that is what I'm trying with the find command. There is probably a cleaner way.

Reply:
Oh right yeah, the `find` finds all files that aren't in our preferred list and removes them. That may be easiest actually because I recall that performing multiple steps in a subdirectory of an action is really verbose.

I think the find command along with maybe capturing the "to keep" directories in a Bash array for readability would be a good way to go.

I'm happy to pick this up tomorrow if you like. I came across this PR because I was hoping to give you a contribution for your review tomorrow but wanted to make sure that I wasn't stepping on your toes haha.

Reply:
Take a look at https://github.com/grafana/writers-toolkit/pull/908/commits/09544788c1d02141d37424170561a0d0c79b96e1 and its run in the grafana repo: https://github.com/grafana/grafana/actions/runs/12081483043/job/33690724463